### PR TITLE
Switch from scraping to downloading undata records

### PR DESCRIPTION
**Pull request by chacalle**
## Describe changes

Switch to downloading X records at a time from data.un.org rather than scraping each separate page. This is much quicker and less prone to error/timeout. 

Also mimics how we were manually downloading files previously.

## What issues are related

Related to #34 

### Packages Repositories

* [X] Have you read the [contributing guidelines](https://github.com/ihmeuw-demographics/packageTemplate/wiki#guide-to-r-package-development) for `ihmeuw-demographics` R packages?
* [X] Have you successfully run `devtools::check()` locally?
* [X] Have you updated or added function (and vignette if applicable) documentation? Did you update the 'man' and 'NAMESPACE' files with `devtools::document()`?
* [X] Have you added in tests for the changes included in the PR?
* [X] Do the changes follow the `ihmeuw-demographics` [code style](https://github.com/ihmeuw-demographics/packageTemplate/wiki/Code-style-guide)?
* [ ] Do the changes need to be immediately included in a new build of [`docker-base`](https://github.com/ihmeuw-demographics/docker-base) or [`docker-internal`](https://github.com/ihmeuw-demographics/docker-internal)? If so follow directions in those repositories to rebuild and redeploy the images.
* [ ] Do the changes require updates to other repositories which use this package? If yes, make the necessary updates in those repos, and consider integration tests for those repositories.

### CODE REVIEW

File: R/get_undata.R [hunk 1]
```diff
@@ -1,114 +1,327 @@
...
+#' @param output_dir \[`character(1)`\]\cr
+#'   Directory to save separate downloaded data files.
+#' @param record_limit \[`numeric(1)`\]\cr
+#'   Upper limit of records/rows that you can download at once.
+#' @param download_sleep \[`numeric(1)`\]\cr
+#'   Seconds to sleep/wait between download calls.
+#' @param base_url \[`character(1)`\]\cr
+#'   The base url of the data.un.org site. Default is 'http://data.un.org/Data.aspx'.
+#' @param downloader_url
+#'   The base url of the data.un.org site download interfact. Default is
```
Comment (hcomfo95) [inline R/get_undata.R]: Nit: change `interfact` to `interface`

File: R/get_undata.R [hunk 2]
```diff
@@ -1,114 +1,327 @@
...
+read_undata_files <- function(fpaths,
+                              n_footnote_col = 2,
+                              data_footnoteid_col = "Value Footnotes",
+                              footnoteid_col = "footnoteSeqID",
+                              footnote_col = "Footnote") {
+
+  if (!requireNamespace("readr", quietly = TRUE)) {
+    stop("Package \"readr\" needed for this function to work. Please install it.",
+         call. = FALSE)
+  }
```
Comment (hcomfo95) [inline R/get_undata.R:232]: Is this needed? I think all singularities we use have the package at this point?

File: R/get_undata.R [hunk 3]
```diff
@@ -1,114 +1,327 @@
...
+        data.table::setnames(new_row, names(new_row), names(footnotes))
+        return(new_row)
+      })
+      concatenated_footnotes <- data.table::rbindlist(concatenated_footnotes)
+      footnotes <- rbind(footnotes, concatenated_footnotes)
+
+      # merge the footnote text onto the data table
+      data.table::setnames(data, data_footnoteid_col, footnoteid_col)
+      data <- merge(data, footnotes, by = footnoteid_col, all.x = TRUE)
+
```
Comment (hcomfo95) [inline R/get_undata.R:280]: Nit: drop the `data.table::`

**Review by hcomfo95 — DISMISSED** [review]

File: R/get_undata.R (hunk 2 above)
Comment (chacalle) [reply R/get_undata.R:232]: This is to start reducing the number of mandatory dependencies we have. Listed here https://github.com/ihmeuw-demographics/demUtils/blob/master/DESCRIPTION and difference explained here https://r-pkgs.org/description.html#dependencies. We were starting to get warnings in this package about too many dependencies which we want to reduce. When we install `demUtils` in our containers, it automatically installs both the imports and suggested packages

**Pull request merged**